outlook/calendar/pkg/commands/searchEvents.go

```diff
@@ -21,19 +22,30 @@ func SearchEvents(ctx context.Context, query string, start, end time.Time) error
 		return fmt.Errorf("failed to create client: %w", err)
 	}
 
-	result, err := graph.SearchEvents(ctx, c, query, start, end)
+	calendars, err := graph.ListCalendars(ctx, c)
 	if err != nil {
-		return fmt.Errorf("failed to search events: %w", err)
+		return fmt.Errorf("failed to list calendars: %w", err)
 	}
 
-	calendarEvents := map[graph.CalendarInfo][]models.Eventable{}
-	for cal, events := range result {
-		if len(events) > 0 {
-			calendarEvents[cal] = events
+	calendarEventsInSubject := map[graph.CalendarInfo][]models.Eventable{}
+	calendarEventsInPreview := map[graph.CalendarInfo][]models.Eventable{}
+	for _, cal := range calendars {
+		result, err := graph.ListCalendarView(ctx, c, cal.ID, cal.Owner, &start, &end)
+		if err != nil {
+			return fmt.Errorf("failed to search events: %w", err)
+		}
+
+		for _, event := range result {
+			if strings.Contains(util.Deref(event.GetSubject()), query) {
+				calendarEventsInSubject[cal] = append(calendarEventsInSubject[cal], event)
+			} else if strings.Contains(util.Deref(event.GetBodyPreview()), query) {
+				calendarEventsInPreview[cal] = append(calendarEventsInPreview[cal], event)
+			}
 		}
 	}
```

Comment:
We used to search the events. This would basically list events but with a search parameter that didn't seem to work very well and sometimes returned syntax errors. At the end of the day, the LLM would get several seemingly irrelevant results.

With this change, we just list all the events within the time window, and then filter them ourselves by checking the subject and body preview (in a case-sensitive way) to see if they contain the query. Still not perfect, but it does better than what we had before.